sklearn/datasets/_base.py

```diff
@@ -215,7 +224,16 @@ def load_files(
     for label, folder in enumerate(folders):
         target_names.append(folder)
         folder_path = join(container_path, folder)
-        documents = [join(folder_path, d) for d in sorted(listdir(folder_path))]
+        files = sorted(listdir(folder_path))
+        if allowed_extensions is not None:
+            allowed_extensions = frozenset(allowed_extensions)
```

Comment:
I think it's a little weird to create the set over and over again through the loop. What do you think of the following?

```python
if allowed_extensions is not None:
    allowed_extensions = frozenset(allowed_extensions)

for label, folder in enumerate(folders):
    ...
    if allowed_extensions is not None:
        documents = [... if os.path.splitext(file)[1] in allowed_extensions]
    else:
        documents = [...]
```

Reply:
Ahhh yeah you're absolutely right, this was a temporary mental lapse on my end (cut-and-paste typo I suppose). Fixed now. Good catch.